internal/engine/image_builder.go

```diff
@@ -25,6 +27,19 @@ func NewImageBuilder(db build.DockerBuilder, custb build.CustomBuilder, updateMo
 	}
 }
 
+func (icb *imageBuilder) CanReuseRef(ctx context.Context, iTarget model.ImageTarget, ref reference.NamedTagged) (bool, error) {
+	switch iTarget.BuildDetails.(type) {
+	case model.DockerBuild:
+		return icb.db.ImageExists(ctx, ref)
+	case model.CustomBuild:
+		// Custom build doesn't have a good way to check if the ref still exists in the image
+		// store, so just assume we can.
+		return true, nil
+	}
+	return false, fmt.Errorf("image %q has no valid buildDetails (neither "+
```

Comment:
(curiosity, not a change request) huh, i'm surprised you `return` outside the `switch` statement instead of using a `default` case. i assume it's b/c go's preference for less indenting controls?

Reply:
It's a bit like cracking a boiled egg on the big-endian side vs cracking an egg on the little-endian side. There's only one right way to do it.